utils/size.util.ts

```diff
@@ -1,27 +1,41 @@
-interface InputHeight {
+interface HeightOption {
   large?: boolean;
   small?: boolean;
   custom?: string;
 }
 
+const BUTTON_HEIGHT = {
+  large: '45',
+  small: '35',
+  default: '40'
+}
+
 const INPUT_HEIGHT = {
   large: '40',
   small: '25',
   default: '35'
 }
 
-export function getInputHeight ({ large, small, custom }: InputHeight): string {
-    if (large) {
-      return INPUT_HEIGHT.large
-    }
+function getElementHeight (elementHeight: Record<string, string>, heightOption: HeightOption): string {
+  if (heightOption.large) {
+    return elementHeight.large
+  }
+  
+  if (heightOption.small) {
+    return elementHeight.small
+  }
 
-    if (small) {
-      return INPUT_HEIGHT.small
-    }
+  if (heightOption.custom) {
+    return heightOption.custom
+  }
 
-    if (custom) {
-      return custom
-    }
+  return elementHeight.default
+}
```

Comment:
Usar "object literals".
Ex: 
```
const { large, small, custom } = heightOption

const optionsHeights =  [large, small, custom]

for(...)
```

Faça um for of, caso não tenha necessariamente que usar o return, use forEatch, lembre de se chegar no final do for tem que retornar o default